### PR TITLE
Drop support for legacy Python 2

### PR DESCRIPTION
**Pull request by hugovk**
## Rationale

Fix https://github.com/SciTools/cartopy/issues/1186.

## Implications

People using Python 2 will not be able to use the most recent release of cartopy.

The Python 2 EOL is coming up on 2020-01-01 and many other projects have pledged to drop support on or before that date: http://python3statement.org

Here's the pip installs for cartopy from PyPI for last month, showing Python 2 use is fairly low:

| category | percent | downloads |
|----------|--------:|----------:|
|      3.6 |  43.75% |     4,011 |
|      2.7 |  22.87% |     2,097 |
|      3.5 |  12.95% |     1,187 |
| null     |  11.19% |     1,026 |
|      3.7 |   8.42% |       772 |
|      3.4 |   0.69% |        63 |
|      3.8 |   0.09% |         8 |
|      2.6 |   0.03% |         3 |
|      3.3 |   0.01% |         1 |
| Total    |         |     9,168 |

Source: `pypistats python_minor --last-month cartopy`

The addition of `python_requires` means, with modern pip and setuptools, users will get the last cartopy which did support Python 2.

### CODE REVIEW

File: lib/cartopy/feature/__init__.py [hunk 1]
```diff
@@ -451,13 +451,13 @@ def __init__(self, wfs, features, **kwargs):
         try:
             from cartopy.io.ogc_clients import WFSGeometrySource
         except ImportError as e:
-            six.raise_from(ImportError(
+            raise ImportError(
                 'WFSFeature requires additional dependencies. If installed '
-                'via pip, try `pip install cartopy[ows]`.\n'), e)
+                'via pip, try `pip install cartopy[ows]`.\n') from e
```
[the remark this answers is not in the document]
Comment (hugovk) [reply lib/cartopy/feature/__init__.py]: This is not valid Python 2.7 syntax.

But this is valid Python 3 syntax.

https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement

**Comment by pelson** [comment]
This is awesome! We can't merge this until *after* v0.18 is released though, and I'm really concerned that this PR will go very stale in that time. Hopefully you were able to run this through 2to3 or something and it was all autogenerated?

Do you want to keep this open (in the knowledge it is likely to go stale), or shall we close and re-open once we have cut v0.18? The latter is my preference, but I'm open to either approaches.

**Comment by hugovk** [comment]
Most of the changes (in the second and third commits) were made using a tool ([pyupgrade](https://github.com/asottile/pyupgrade)), so it'll be easy to redo this after 0.18 is out.

I'd probably prefer to keep it open so it's visible and not forgotten, but don't have a strong preference. Do whichever fits you feel is best for this project :)

**Comment by pelson** [comment]
In which case I will go ahead and close this. We have the issue to track the problem, and it links directly to this PR so we can easily enough find this again.

Thanks again for doing this - it was useful to see the kinds of things we are going to be able to remove once py2 is dropped, not to mention all of the features we can start to benefit from.

**Comment by QuLogic** [comment]
Hey, so I forgot about this PR and put in #1516 and #1517, but they do not cover everything that's here. If you want to re-run the tools and put together a new PR, feel free to do so.

**Comment by hugovk** [comment]
Heh, will do, thanks!

**Comment by hugovk** [comment]
Please see PR #1540.

**Pull request closed without merging**